src/components/annotations/draw.js

```diff
@@ -118,6 +118,19 @@ function drawOne(gd, index) {
         .call(Color.stroke, options.bordercolor)
         .call(Color.fill, options.bgcolor);
 
+    var isSizeConstrained = options.width || options.height;
+
+    var annClipID = 'clip' + fullLayout._uid + '_ann' + index;
+    var annTextClip = fullLayout._defs.select('.clips')
+        .selectAll('#' + annClipID)
+        .data(isSizeConstrained ? [0] : []);
+
+    annTextClip.enter().append('clipPath')
+        .classed('annclip', true)
+        .attr('id', annClipID)
+      .append('rect');
+    annTextClip.exit().remove();
```

Comment:
actually was easy to remove the unneeded clip paths after all -> https://github.com/plotly/plotly.js/pull/1551/commits/b21154bf138e4a1e49019a5ac72a1e7aca9e54bf